### PR TITLE
edits + restructuring

### PR DESCRIPTION
**Pull request by lbjay**
I went a little overboard. Some of these changes are meant to be illustrative of various python idioms and techniques.

* get the `.env` vars up top, use as arg defaults
* moved arg parsing into the `__main__` block
* allow output to stdout
* added a `--date` arg
* refactored so that each run gets either "live" or "past" meetings (not both)
* added some exception handling

### CODE REVIEW

File: meetings.py [hunk 1]
```diff
@@ -10,80 +12,94 @@
 from dotenv import load_dotenv
 load_dotenv(join(dirname(__file__), '.env'))
 
+KEY = getenv('ZOOM_KEY')
```
Comment (lbjay) [inline meetings.py:15]: best to set these kinds of things at the global module scope

File: meetings.py [hunk 2]
```diff
@@ -10,80 +12,94 @@
...
-def main():
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--key", help="zoom api key")
-    parser.add_argument("--secret", help="zoom api secret")
-    parser.add_argument("--destination", help="destination filename")
-    args = parser.parse_args()
-
-    if args.key is None:
-        KEY = getenv('ZOOM_KEY')
+class ZoomApiException(Exception):
```
Comment (lbjay) [inline meetings.py:20]: example use case for custom exception

File: meetings.py [hunk 3]
```diff
@@ -10,80 +12,94 @@
...
-        'type': "2",
-        'page_number': 1,
-        'page_size': "300",
+    params = {
+        'from': date,
+        'to': date,
+        'type': MEETING_TYPES[type],
+        'api_key': key,
+        'api_secret': secret,
+        'page_size': 100,
```
Comment (lbjay) [inline meetings.py:31]: looked like 100 was the max allowed by the api

File: meetings.py [hunk 4]
```diff
@@ -10,80 +12,94 @@
...
-
-            if r.status_code != requests.codes.ok:
-                print("Failed with status code:", r.status_code, "reason:", r.reason)
-                return
+    meetings = []
+    while True:
+
+        r = requests.post(url=API_BASE_URL + "/metrics/meetings", data=params)
+        r.raise_for_status()
+        response = r.json()
```
Comment (lbjay) [inline meetings.py:40]: requests response objects have this convenience method when the content-type is json

File: meetings.py [hunk 5]
```diff
@@ -10,80 +12,94 @@
...
+
+        if 'error' in response.keys():
```
Comment (lbjay) [inline meetings.py:42]: i'm curious what kinds of errors does the api return in this way?

File: meetings.py [hunk 6]
```diff
@@ -10,80 +12,94 @@
...
+            raise ZoomApiException(response['error'])
+
+        for meeting in response['meetings']:
+            meeting['type'] = response['type']
```
Comment (lbjay) [inline meetings.py:46]: i think this wasn't working but didn't look too closely. the meeting data seemed to always have a "type" value like "Past Meetings" even after modifying it here.

**Review by nataliemona — APPROVED** [review]

**Pull request merged**